### PR TITLE
How Java Virtual Machine Works

### LINKED ISSUE #3496: How Java Virtual Machine Works
### Proposed title of article
How Java Virtual Machine Works 

### Introduction paragraph (2-3 paragraphs):
The JVM (Java Virtual Machine) is a run-time engine that allows Java programs to run. The JVM is the program that executes the main function in java programming. JVM is a component of JRE (Java Runtime Environment). The compiler in other programming languages generates machine code for a specific system. The Java compiler, on the other hand, generates code for the Java Virtual Machine.

At the end of this article, the reader shall understand how a java virtual machine works.

### Key takeaways: 
-How does JVM execution engine work?
-How JVM Memory is used.
-The Java Native Interface (JNI).

### References: 
N/A

### PR DESCRIPTION
**Pull request by Geofreytech**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3496 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Some few words or sentences were flagged in the plagiarism check. Could you @Geofreytech resolve them
[How Java Virtual Machine Works #3568.pdf](https://github.com/section-engineering-education/engineering-education/files/7072126/How.Java.Virtual.Machine.Works.3568.pdf)

**Comment by Geofreytech** [comment]
Done, @WanjaMIKE

**Comment by Geofreytech** [comment]
May i ask about the progress of this article, please?

**Comment by lalith1403** [comment]
The name of the folder is /How-Java-Virtual-Machine-Works/... Directory names should not have uppercase letters. Can you change it to lowercase letters?

**Comment by Geofreytech** [comment]
@lalith1403 
I am really sorry I am unable to.

**Comment by lalith1403** [comment]
You will have to pull the changes to your local computer, rename the folder, and commit the changes and push the changes.

**Comment by Geofreytech** [comment]
> You will have to pull the changes to your local computer, rename the folder, and commit the changes and push the changes.

Okay.

**Comment by Geofreytech** [comment]
@lalith1403, I have renamed the folder. Please have a look at my last commit.

File: content/articles/how-java-virtual-machine-works/index.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    public String getName() { return name; }
+    public void setName(String name) { this.name = name; }
+    public int getRoll_no() { return roll_No; }
+    public void setRoll_no(int roll_no)
+    {
+        this.roll_No = roll_no;
+    }
+}
+```
+**Output**
```
Comment (lalith1403) [inline content/articles/how-java-virtual-machine-works/index.md:85]: Explain what you are trying to show with the code.

**Comment by Geofreytech** [comment]
@lalith1403 I have corrected. Thank you.

**Comment by hectorkambow** [comment]
@Geofreytech 

Good afternoon and thank you for submitting your topic suggestion.
 
Not a problem - it seems this PR is still quite a bit aways from being completed. 
 
As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
 
We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 
 
👍 


cc @WanjaMIKE 
cc @lalith1403

**Pull request closed without merging**